### ISSUE TITLE
Using body_listener is conflicting with the symfony user argument resolver, resulting in a 400

### ISSUE THREAD

**Issue by norbert-yoimo** [FriendsOfSymfony/FOSRestBundle#1659]
If you have the body_listener, enabled, and try to use the [$user argument resolver](http://symfony.com/blog/new-in-symfony-3-2-user-value-resolver-for-controllers) in symfony, like this:

```php
namespace MyBundle\Controller;
use FOS\RestBundle\Controller\FOSRestController;
use FOS\RestBundle\Controller\Annotations\View as FOSView;
use FOS\RestBundle\Controller\Annotations\Get;
use Symfony\Component\Security\Core\User\UserInterface;

class MyController extends FOSRestController {

    /**
     * @FOSView
     * @Get("/team/X")
     */
    public function getXAction(UserInterface $user) {
    }
}
```

You will always get a `400 Bad Request` response from this route, with the body:

```
{
    "code": 400,
    "message": "Could not decode JSON, syntax error - malformed JSON."
}
```

Looking into the stacktrace, there is a call to:

```
at Serializer ->deserialize ('', 'Symfony\\Component\\Security\\Core\\User\\UserInterface', 'json', object(DeserializationContext)) 
in vendor/friendsofsymfony/rest-bundle/Serializer/JMSSerializerAdapter.php at line 60
```

Which makes me believe that the FOSRestBundle body_listener is trying to deserialize an empty string as the user object, instead of letting the symfony argument resolver take care of it.

**Comment by GuilhemN** [comment]
Can you check that https://github.com/FriendsOfSymfony/FOSRestBundle/issues/1669 fixes your issue?

**Comment by norbert-yoimo** [comment]
Sorry I haven't yet had time to take a look, I'm still keeping it on my task list

**Comment by Preclowski** [comment]
Problem still exists

**Comment by anonim1133** [comment]
It's still pain in the ass. #1669 fixes it, but does not exist in newer branches.

**Comment by xabbuh** [comment]
fixed in #1669
